src/CMakeLists.txt

```diff
@@ -1,72 +1,45 @@
-include(DefaultPathsMRCPP)
-
 add_library(mrcpp STATIC 
-    ABGVCalculator.cpp
-    ABGVOperator.cpp
-    BandWidth.cpp
-    BoundingBox.cpp
-    BoysFunction.cpp
-    ConvolutionCalculator.cpp
-    ConvolutionOperator.cpp
-    CopyAdaptor.cpp
-    CrossCorrelation.cpp
-    CrossCorrelationCache.cpp
-    CrossCorrelationCalculator.cpp
-    DerivativeCalculator.cpp
-    FilterCache.cpp
-    FunctionNode.cpp
-    FunctionTree.cpp
-    FunctionTreeVector.cpp
-    GaussExp.cpp
-    GaussFunc.cpp
-    GaussPoly.cpp
-    GaussQuadrature.cpp
-    Gaussian.cpp
-    GenNode.cpp
-    grid.cpp
-    HelmholtzKernel.cpp
-    HelmholtzOperator.cpp
-    HilbertPath.cpp
-    InterpolatingBasis.cpp
-    LegendreBasis.cpp
-    LegendrePoly.cpp
-    MWFilter.cpp
-    MWNode.cpp
-    MWOperator.cpp
-    MWTree.cpp
-    project.cpp
-    apply.cpp
-    add.cpp
-    multiply.cpp
-    MathUtils.cpp
-    MultiResolutionAnalysis.cpp
-    NodeBox.cpp
-    ObjectCache.cpp
-    OperatorNode.cpp
-    OperatorStatistics.cpp
-    OperatorTree.cpp
-    PHCalculator.cpp
-    PHOperator.cpp
-    Plotter.cpp
-    PoissonKernel.cpp
-    PoissonOperator.cpp
-    Polynomial.cpp
-    ProjectedNode.cpp
-    ProjectionCalculator.cpp
-    QuadratureCache.cpp
-    RepresentableFunction.cpp
-    ScalingBasis.cpp
-    SerialFunctionTree.cpp
-    SerialOperatorTree.cpp
-    SerialTree.cpp
-    Printer.cpp
-    Timer.cpp
-    TreeBuilder.cpp
-    TreeIterator.cpp
     parallel_mpi.cpp
 )
 
-install(TARGETS mrcpp
-        ARCHIVE DESTINATION lib
-        CONFIGURATIONS Debug-devel Release-devel
-        )
+add_subdirectory(mwcore)
+add_subdirectory(mwutils)
+add_subdirectory(mwtrees)
+add_subdirectory(mwbuilders)
+add_subdirectory(mwfunctions)
+add_subdirectory(mwoperators)
+
+target_include_directories(mrcpp PRIVATE
+    ${CMAKE_BINARY_DIR}
+    ${CMAKE_SOURCE_DIR}/src/mwcore
+    ${CMAKE_SOURCE_DIR}/src/mwutils
+    ${CMAKE_SOURCE_DIR}/src/mwtrees
+    ${CMAKE_SOURCE_DIR}/src/mwbuilders
+    ${CMAKE_SOURCE_DIR}/src/mwfunctions
+    ${CMAKE_SOURCE_DIR}/src/mwoperators
+    )
+
+install(TARGETS mrcpp ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+
+list(APPEND mrcpp_h 
+    constants.h
+    eigen_disable_warnings.h
+    macros.h
+    mrcpp_declarations.h
+    parallel_mpi.h
+    parallel_omp.h
+    )                                                       
+                                                                                 
+# Sets install directory for all the headers in the list                         
+foreach(_header ${mrcpp_h})                                                 
```

Comment:
I think these files should go under `${CMAKE_INSTALL_INCLUDEDIR]/${PROJECT_NAME}` You might remove the loop using the `REGEX` option to `file(COPY` and `install` but you'll have to experiment a bit since I've never tried it.